raft/src/main/java/org/apache/kafka/raft/internals/KafkaRaftMetrics.java

```diff
@@ -63,9 +63,12 @@ public KafkaRaftMetrics(Metrics metrics, String metricGrpPrefix, QuorumState sta
         this.numUnknownVoterConnections = 0;
         this.logEndOffset = new OffsetAndEpoch(0L, 0);
 
-        this.currentStateMetricName = metrics.metricName("current-state", metricGroupName, "The current state of this member; possible values are leader, candidate, voted, follower, unattached");
+        this.currentStateMetricName = metrics.metricName("current-state", metricGroupName, "The current state of this member; possible values are leader, candidate, voted, follower, unattached, observer");
         Gauge<String> stateProvider = (mConfig, currentTimeMs) -> {
-            if (state.isLeader()) {
+            // a broker, as not being a voter, is always an observer
+            if (state.isObserver()) {
+                return "observer";
```

Comment:
@ppatierno , I think this check is not correct, because the `isObserver` is checking if this node a "voter" or not. In another word, `observer` is not actually a kind of `state`. Thus, it won't care about if it's `unattached` or not. 

I'm thinking, the `observer` state is actually another term of `follower`, just it doesn't join voting process. So, we can change the logic like this:

```
if (state.isLeader()) {
    return "leader";
} else if (state.isCandidate()) {
    return "candidate";
} else if (state.isVoted()) {
    return "voted";
} else if (state.isFollower()) {
    // only show "observer" when it's in "follower" state
    if (state.isObserver()) {
        return "observer";
    } else {
        return "follower";
    }
} else {
    return "unattached";
}
```

I think that makes more sense. WDYT?

Reply:
> In another word, observer is not actually a kind of state

My doubt was about adding a new state at the beginning but then I thought it was not worth in the end for how it's used somewhere else.

> I'm thinking, the observer state is more like an another term of follower, just it doesn't join voting process.

Yes, agree. So I think your proposal makes more sense, I will apply it.